### PR TITLE
feat: [OIP-442]: Fix multiple issues observed during testing baseline…

### PR DESCRIPTION
**Pull request by naveen-harness**
… based verification.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/48291)

### CODE REVIEW

**Comment by naveen-harness** [comment]
trigger messagecheck

**Comment by naveen-harness** [comment]
trigger messagecheck

**Comment by naveen-harness** [comment]
trigger uts

**Comment by naveen-harness** [comment]
trigger uts

**Comment by naveen-harness** [comment]
trigger uts

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/beans/TestVerificationJobSpec.java [hunk 1]
```diff
@@ -29,6 +32,13 @@ public String getType() {
     return "LoadTest";
   }
 
+  @ApiModelProperty(dataType = SwaggerConstants.STRING_CLASSPATH, value = "Possible values: [LAST, PINNED]")
+  ParameterField<String> baseline;
+
+  public ParameterField<String> getBaseline() {
+    return baseline;
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/beans/TestVerificationJobSpec.java:39]: I wonder if the usage of this field backward compatible... can we refactor this method to return value as LAST if the set value is null?

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/DeploymentLogClusterStateExecutor.java [hunk 2]
```diff
@@ -53,6 +59,14 @@ public AnalysisState handleTransition(DeploymentLogClusterState analysisState) {
...
         return deploymentLogClusterState;
       case L2:
+        VerificationJobInstance verificationJobInstance = verificationJobInstanceService.getVerificationJobInstance(
+            analysisState.getInputs().getVerificationJobInstanceId());
+        if (verificationJobInstance.getResolvedJob().getType() == VerificationJobType.TEST) {
+          DeploymentLogAnalysisState deploymentLogAnalysisState = DeploymentLogAnalysisState.builder().build();
+          deploymentLogAnalysisState.setInputs(analysisState.getInputs());
+          deploymentLogAnalysisState.setStatus(AnalysisStatus.CREATED);
+          return deploymentLogAnalysisState;
+        }
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/DeploymentLogClusterStateExecutor.java:69]: nit: We should use else {...} for easier reading

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/verificationjob/services/impl/VerificationJobInstanceServiceImpl.java [hunk 3]
```diff
@@ -457,13 +457,12 @@ public Baseline pinOrUnpinBaseline(VerifyStepPathParams verifyStepPathParams, bo
                     baselineVerificationJobInstance.getUuid(), verifyStepPathParams.getVerifyStepExecutionId())
                 && isBaseline))) {
       // Duration can't be negative
-      Duration updatedTtl = CVConstants.MAX_DATA_RETENTION_DURATION.plus(
-          Duration.ofMillis(baselineVerificationJobInstance.getCreatedAt()));
+      Instant instant = Instant.ofEpochMilli(baselineVerificationJobInstance.getCreatedAt());
+      Instant updatedInstant = instant.plus(CVConstants.MAX_DATA_RETENTION_DURATION);
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/verificationjob/services/impl/VerificationJobInstanceServiceImpl.java:461]: nit: We can refactor these fields to baselineVerificationCreatedAt and baselineVerificationValidUntil

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2021 Harness Inc. All rights reserved.
```
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java:2]: nit: Year

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java
Comment (dhruv-harness) [inline srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java:1]: Lets add UT for handling the transition when expected state is DeploymentLogHostSamplingState

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/beans/TestVerificationJobSpec.java (hunk 1 above)
Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/cdng/beans/TestVerificationJobSpec.java:42]: Usage of this field IS backward compatible, I've taken care of it TestVerificationJob#resolveAdditionsFields, and in dev testing I didn't observe any backward compatibility issues, however fixed it here as well just to be fail safe. Thanks!

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/DeploymentLogClusterStateExecutor.java (hunk 2 above)
Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/statemachine/services/api/DeploymentLogClusterStateExecutor.java:74]: Done

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/verificationjob/services/impl/VerificationJobInstanceServiceImpl.java (hunk 3 above)
Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/verificationjob/services/impl/VerificationJobInstanceServiceImpl.java]: Done

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java (hunk 4 above)
Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java]: Done

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/verificationjob/services/impl/VerificationJobInstanceServiceImplTest.java [hunk 5]
```diff
@@ -1037,6 +1037,9 @@ public void testPinOrUnpinBaseline_pinBaseline_success() {
     VerificationJobInstance unpinnedBaselineVerificationJobInstance =
         verificationJobInstanceService.getVerificationJobInstance(baselineVerificationJobInstanceId);
     assertThat(unpinnedBaselineVerificationJobInstance.getIsBaseline()).isFalse();
+    Instant instant = Instant.ofEpochMilli(baselineVerificationJobInstance.getCreatedAt());
```
[the remark this answers is not in the document]
Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/verificationjob/services/impl/VerificationJobInstanceServiceImplTest.java]: Done

File: srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java
Comment (naveen-harness) [reply srm-service/modules/cv-nextgen-service/service/src/test/java/io/harness/cvng/statemachine/services/impl/DeploymentLogClusterStateExecutorTest.java:1]: Done

**Comment by naveen-harness** [comment]
trigger uts

**Comment by naveen-harness** [comment]
trigger uts

**Comment by naveen-harness** [comment]
trigger uts

**Review by dhruv-harness — APPROVED** [review]

**Pull request merged**